File: src/main/java/jp/aoichaan0513/A_TosoGame_Live/Commands/Command/T.java
package jp.aoichaan0513.A_TosoGame_Live.Commands.Command;

import jp.aoichaan0513.A_TosoGame_Live.API.MainAPI;
import jp.aoichaan0513.A_TosoGame_Live.API.Manager.GameManager;
import jp.aoichaan0513.A_TosoGame_Live.API.Scoreboard.Teams;
import jp.aoichaan0513.A_TosoGame_Live.Commands.ICommand;
import jp.aoichaan0513.A_TosoGame_Live.Main;
import org.bukkit.ChatColor;
import org.bukkit.command.BlockCommandSender;
import org.bukkit.command.Command;
import org.bukkit.command.ConsoleCommandSender;
import org.bukkit.entity.Player;

import java.util.List;

public class T extends ICommand {

    public T(String name) {
        super(name);
    }

    @Override
    public void onPlayerCommand(Player sp, Command cmd, String label, String[] args) {
        if (Tuho.num > 0) {
            if (GameManager.isGame(GameManager.GameState.GAME)) {
                if (Teams.hasJoinedTeam(Teams.OnlineTeam.TOSO_JAIL, sp)) {
                    if (!Main.tuhoShuffleSet.contains(sp.getUniqueId()))
                        Main.tuhoShuffleSet.add(sp.getUniqueId());
                    else
                        Main.tuhoShuffleSet.remove(sp.getUniqueId());

                    sp.sendMessage(MainAPI.getPrefix(MainAPI.PrefixType.WARNING) + ChatColor.UNDERLINE + "通報部隊募集" + ChatColor.GOLD + ChatColor.UNDERLINE + (Main.tuhoShuffleSet.contains(sp.getUniqueId()) ? "に応募" : "の応募をキャンセル") + ChatColor.RESET + ChatColor.YELLOW + "しました。");
                    return;
                }
                MainAPI.sendMessage(sp, MainAPI.ErrorMessage.PERMISSIONS_TEAM_JAIL);
                return;
            } else {
                if (!Main.tuhoShuffleSet.contains(sp.getUniqueId()))
                    Main.tuhoShuffleSet.add(sp.getUniqueId());
                else
                    Main.tuhoShuffleSet.remove(sp.getUniqueId());

                sp.sendMessage(MainAPI.getPrefix(MainAPI.PrefixType.WARNING) + ChatColor.UNDERLINE + "通報部隊募集" + ChatColor.GOLD + ChatColor.UNDERLINE + (Main.tuhoShuffleSet.contains(sp.getUniqueId()) ? "に応募" : "の応募をキャンセル") + ChatColor.RESET + ChatColor.YELLOW + "しました。");
                return;
            }
        }
        sp.sendMessage(MainAPI.getPrefix(MainAPI.PrefixType.ERROR) + "通報部隊を募集していないため実行できません。");
        return;
    }

    @Override
    public void onBlockCommand(BlockCommandSender bs, Command cmd, String label, String[] args) {
        MainAPI.sendMessage(bs, MainAPI.ErrorMessage.NOT_PLAYER);
        return;
    }

    @Override
    public void onConsoleCommand(ConsoleCommandSender cs, Command cmd, String label, String[] args) {
        MainAPI.sendMessage(cs, MainAPI.ErrorMessage.NOT_PLAYER);
        return;
    }

    @Override
    public List<String> onPlayerTabComplete(Player sp, Command cmd, String alias, String[] args) {
        return null;
    }

    @Override
    public List<String> onBlockTabComplete(BlockCommandSender bs, Command cmd, String alias, String[] args) {
        return null;
    }

    @Override
    public List<String> onConsoleTabComplete(ConsoleCommandSender cs, Command cmd, String alias, String[] args) {
        return null;
    }
}

File: src/main/kotlin/jp/aoichaan0513/A_TosoGame_Live/Commands/Command/T.kt
package jp.aoichaan0513.A_TosoGame_Live.Commands.Command

import jp.aoichaan0513.A_TosoGame_Live.API.MainAPI
import jp.aoichaan0513.A_TosoGame_Live.API.MainAPI.ErrorMessage
import jp.aoichaan0513.A_TosoGame_Live.API.MainAPI.PrefixType
import jp.aoichaan0513.A_TosoGame_Live.API.Manager.GameManager
import jp.aoichaan0513.A_TosoGame_Live.API.Manager.GameManager.GameState
import jp.aoichaan0513.A_TosoGame_Live.API.Scoreboard.Teams
import jp.aoichaan0513.A_TosoGame_Live.API.Scoreboard.Teams.OnlineTeam
import jp.aoichaan0513.A_TosoGame_Live.Commands.ICommand
import jp.aoichaan0513.A_TosoGame_Live.Main
import org.bukkit.ChatColor
import org.bukkit.command.BlockCommandSender
import org.bukkit.command.Command
import org.bukkit.command.ConsoleCommandSender
import org.bukkit.entity.Player

class T(name: String) : ICommand(name) {

    override fun onPlayerCommand(sp: Player, cmd: Command, label: String, args: Array<String>) {
        if (Team.isTuhoRandom) {
            if (GameManager.isGame(GameState.GAME)) {
                if (Teams.hasJoinedTeam(OnlineTeam.TOSO_JAIL, sp)) {
                    if (!Main.tuhoShuffleSet.contains(sp.uniqueId)) Main.tuhoShuffleSet.add(sp.uniqueId) else Main.tuhoShuffleSet.remove(sp.uniqueId)
                    sp.sendMessage("${MainAPI.getPrefix(PrefixType.WARNING)}${ChatColor.UNDERLINE}通報部隊募集${ChatColor.GOLD}${ChatColor.UNDERLINE}${if (Main.hunterShuffleSet.contains(sp.uniqueId)) "に応募" else "の応募をキャンセル"}${ChatColor.RESET}${ChatColor.YELLOW}しました。")
                    return
                }
                MainAPI.sendMessage(sp, ErrorMessage.PERMISSIONS_TEAM_JAIL)
                return
            } else {
                if (!Main.tuhoShuffleSet.contains(sp.uniqueId)) Main.tuhoShuffleSet.add(sp.uniqueId) else Main.tuhoShuffleSet.remove(sp.uniqueId)
                sp.sendMessage("${MainAPI.getPrefix(PrefixType.WARNING)}${ChatColor.UNDERLINE}通報部隊募集${ChatColor.GOLD}${ChatColor.UNDERLINE}${if (Main.hunterShuffleSet.contains(sp.uniqueId)) "に応募" else "の応募をキャンセル"}${ChatColor.RESET}${ChatColor.YELLOW}しました。")
                return
            }
        }
        sp.sendMessage("${MainAPI.getPrefix(PrefixType.ERROR)}通報部隊を募集していないため実行できません。")
        return
    }

    override fun onBlockCommand(bs: BlockCommandSender, cmd: Command, label: String, args: Array<String>) {
        MainAPI.sendMessage(bs, ErrorMessage.NOT_PLAYER)
        return
    }

    override fun onConsoleCommand(cs: ConsoleCommandSender, cmd: Command, label: String, args: Array<String>) {
        MainAPI.sendMessage(cs, ErrorMessage.NOT_PLAYER)
        return
    }

    override fun onPlayerTabComplete(sp: Player, cmd: Command, alias: String, args: Array<String>): List<String>? {
        return null
    }

    override fun onBlockTabComplete(bs: BlockCommandSender, cmd: Command, alias: String, args: Array<String>): List<String>? {
        return null
    }

    override fun onConsoleTabComplete(cs: ConsoleCommandSender, cmd: Command, alias: String, args: Array<String>): List<String>? {
        return null
    }
}